azkaban-common/src/main/java/azkaban/project/ArchiveUnthinner.java

```diff
@@ -272,6 +276,8 @@ private void updateValidationStatuses(final Set<? extends Dependency> untouchedD
         .keySet()
         .stream()
         .filter(d -> validationStatuses.get(d) == status)
+        .filter(Objects::nonNull) // Filter out null since the Set should contain Dependency
```

Comment:
nonNull should be above filter for validationStatuses

Reply:
Changed the order

Reply:
@li-gsalia the order appears same. Did you push the change?

Reply:
Pushed the change now, laptop had issues yesterday.

Reply:
Pushed the change now, had some laptop issues yesterday.